### PR TITLE
correspondence between cardano-cli and cardano-addresses keys

### PR DESCRIPTION
**Pull request by paweljakubas**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -194,6 +194,67 @@ addr_test1wqqggtajwkxjgf58v452jz6jl87lt32w3mhez5hd7xz6hugp80tta
...
+$ cardano-address key child 1852H/1815H/0H/0/0 < root.xprv > key.xsk
+addr_xsk1kzl5vgev0u843tfnxqcwg0lmaf7zhdhczddaqhas6dp6m6z98302e3avp8mhu94kxkpj2gss064f74km3rrptafh4fsztekz8k5c469shcvx35wrdmus3xemp984lcwhs0jdtl4pfcsrfspe00h9pej6rg8drvcv
+
+Create signing key using cardano-cli
+$ cardano-cli key convert-cardano-address-key --shelley-payment-key --signing-key-file key.xsk --out-file key.skey
+{
+    "type": "PaymentExtendedSigningKeyShelley_ed25519_bip32",
+    "description": "",
+    "cborHex": "5880b0bf46232c7f0f58ad333030e43ffbea7c2bb6f8135bd05fb0d343ade8453c5eacc7ac09f77e16b635832522107eaa9f56db88c615f537aa6025e6c23da98ae8fbbbf6410e24532f35e9279febb085d2cc05b3b2ada1df77ea1951eb694f3834b0be1868d1c36ef9089b3b094f5fe1d783e4d5fea14e2034c0397bee50e65a1a"
+}
```
Comment (piotr-iohk) [inline README.md:219]: It is suggested here that `cat key.xsk | bech32` is the same output as `cardano-cli key convert-cardano-address-key --shelley-payment-key --signing-key-file key.xsk --out-file key.skey` (minus `5880` prefix)... at least that's how I understood it.

However it looks like `PaymentExtendedSigningKeyShelley_ed25519_bip32` contains:
 - `5880` prefix
 - `b0bf46232c7f0f58ad333030e43ffbea7c2bb6f8135bd05fb0d343ade8453c5eacc7ac09f77e16b635832522107eaa9f56db88c615f537aa6025e6c23da98ae8` signing key
 -  `fbbbf6410e24532f35e9279febb085d2cc05b3b2ada1df77ea1951eb694f3834` verification key
 - `b0be1868d1c36ef9089b3b094f5fe1d783e4d5fea14e2034c0397bee50e65a1a` chain code

Which is not the same as the output of:
```
$ cardano-address key child 1852H/1815H/0H/0/0 < root.xprv > key.xsk
$ cat key.xsk | bech32
b0bf46232c7f0f58ad333030e43ffbea7c2bb6f8135bd05fb0d343ade8453c5eacc7ac09f77e16b635832522107eaa9f56db88c615f537aa6025e6c23da98ae8b0be1868d1c36ef9089b3b094f5fe1d783e4d5fea14e2034c0397bee50e65a1a
```
Which consists of:
 - `b0bf46232c7f0f58ad333030e43ffbea7c2bb6f8135bd05fb0d343ade8453c5eacc7ac09f77e16b635832522107eaa9f56db88c615f537aa6025e6c23da98ae8` signing key
 - `b0be1868d1c36ef9089b3b094f5fe1d783e4d5fea14e2034c0397bee50e65a1a` chain code

Correct me if I'm wrong pls.

Comment (paweljakubas) [reply README.md:219]: of course you are right -> corrected!

File: README.md [hunk 2]
```diff
@@ -194,6 +194,93 @@ addr_test1wqqggtajwkxjgf58v452jz6jl87lt32w3mhez5hd7xz6hugp80tta
...
+    "type": "PaymentExtendedVerificationKeyShelley_ed25519_bip32",
+    "description": "",
+    "cborHex": "5840fbbbf6410e24532f35e9279febb085d2cc05b3b2ada1df77ea1951eb694f3834b0be1868d1c36ef9089b3b094f5fe1d783e4d5fea14e2034c0397bee50e65a1a"
+}
+The cborhex here contains of 3 parts:
+1. prefix 5840 - bytestring of 64 bytes
+2. verification key (32 bytes) - fbbbf6410e24532f35e9279febb085d2cc05b3b2ada1df77ea1951eb694f3834
+3. chain code (32 bytes) - b0be1868d1c36ef9089b3b094f5fe1d783e4d5fea14e2034c0397bee50e65a1a
+
+Rule for prefixes: CBOR-encoded bytestring (which is what the 58 identifies) of a specific size (80 means 128 bytes, whereas 40 means 64 bytes, 20 means 32 bytes)
```
Comment (piotr-iohk) [inline README.md:239]:
```suggestion
Rule for prefixes: 
 - CBOR-encoded bytestring (which is what the 58 identifies) 
 - size (80 means 128 bytes, whereas 40 means 64 bytes, 20 means 32 bytes)
```
Suggestion for breaking the line somehow for reader's convenience.

File: README.md [hunk 3]
```diff
@@ -194,6 +194,93 @@ addr_test1wqqggtajwkxjgf58v452jz6jl87lt32w3mhez5hd7xz6hugp80tta
...
+Within cardano-addresses one can get cborhex of verification key (without chain code)
+$ cardano-address key public --without-chain-code < key.xsk | bech32
+fbbbf6410e24532f35e9279febb085d2cc05b3b2ada1df77ea1951eb694f3834
+(just verification key)
+
+Then, we can get compute hash (but here we need to use without chain code):
+$ cardano-address key public --without-chain-code < key.xsk | cardano-address key hash | bech32
+0185545935760c5e370d01e6f4fedbb89b7fd79e115f2837cfab9ea8
+
+Now, we can deal with non-extended keys, and construct non-extended private key for account ix=0H, role=0 and address ix=0
```
Comment (piotr-iohk) [inline README.md:272]:
```suggestion
Now, we can deal with non-extended keys and construct non-extended private key for:
 - account ix=0H, 
 - role=0,
 - address ix=0
```

**Review by piotr-iohk — APPROVED** [review]
LGTM. Very detailed and informative!

Small suggestions here and there for breaking comment lines for reader's convenience as some comments go off screen in the README.

**Pull request merged**